blockchain/blockchain.go

```diff
@@ -99,30 +100,18 @@ type Blockchain interface {
 	BlockFooterByHeight(height uint64) (*block.Footer, error)
 	// BlockFooterByHash return block footer by hash
 	BlockFooterByHash(h hash.Hash256) (*block.Footer, error)
-	// GetTotalActions returns the total number of actions
-	GetTotalActions() (uint64, error)
-	// GetNumActions returns the number of actions in certain block
-	GetNumActions(height uint64) (uint64, error)
-	// GetTranferAmount returns the transfer amount
-	GetTranferAmount(height uint64) (*big.Int, error)
 	// GetReceiptByActionHash returns the receipt by action hash
 	GetReceiptByActionHash(h hash.Hash256) (*action.Receipt, error)
-	// GetActionsFromIndex returns action hash from index
-	GetActionsFromIndex(uint64, uint64) ([][]byte, error)
-	// GetActionsByAddress returns actions by address
-	GetActionsByAddress(string, uint64, uint64) ([][]byte, error)
-	// GetActionCountByAddress returns action count by address
-	GetActionCountByAddress(address string) (uint64, error)
 	// GetActionByActionHash returns action by action hash
 	GetActionByActionHash(h hash.Hash256) (action.SealedEnvelope, error)
-	// GetBlockHashByActionHash returns Block hash by action hash
-	GetBlockHashByActionHash(h hash.Hash256) (hash.Hash256, error)
 	// GetReceiptsByHeight returns action receipts by block height
 	GetReceiptsByHeight(height uint64) ([]*action.Receipt, error)
 	// GetFactory returns the state factory
 	GetFactory() factory.Factory
 	// GetBlockDAO returns the block DAO
 	GetBlockDAO() blockdao.BlockDAO
+	// GetIndexer returns the chain indexer
+	GetIndexer() blockindex.Indexer
```

Comment:
this is duplicated with GetBlockDao. We may merge them into one as
```Indexer() blockchain.Indexer```

Reply:
After rethinking about it, the API of blockchain may not need to have interfaces like GetFactory and GetBlockDao. Our definition of Blockchain is a mix of Blockchain and BlockchainBuilder, that's why we can StateFactoryOption, BlockDaoOption and etc.